### PR TITLE
setup container env for linux xdp

### PR DESCRIPTION
**Pull request by ami-GS**
## Description

Linux xdp dependencies

## Testing

N/A

## Documentation

not officially supported yet

### CODE REVIEW

**Comment by ami-GS** [comment]
For some reason gcc-multilib is uninstalled at some point

**Review by nibanks — DISMISSED** [review]

**Comment by csujedihy** [comment]
> For some reason gcc-multilib is uninstalled at some point

Didn't the docker build fail because it couldn't find ethertool?

And why are you still installing non-build related stuff on this docker? e.g. iproute2, iptables and ethertool

File: .docker/ubuntu-22.04/Dockerfile [hunk 1]
```diff
@@ -50,6 +64,12 @@ RUN wget -q https://packages.microsoft.com/config/ubuntu/18.04/packages-microsof
         powershell \
   && rm -rf /var/lib/apt/lists/*
 
+RUN wget -q https://github.com/libbpf/bpftool/releases/download/v7.2.0/bpftool-v7.2.0-amd64.tar.gz -O bpftool-amd64.tar.gz && \
```
Comment (csujedihy) [inline .docker/ubuntu-22.04/Dockerfile:67]: Is bpftool needed for **building** the project?

File: .docker/ubuntu-22.04/Dockerfile [hunk 2]
```diff
@@ -4,7 +4,9 @@ LABEL org.opencontainers.image.source https://github.com/microsoft/msquic
 
 ENV DEBIAN_FRONTEND=noninteractive
 
-RUN apt-get update && apt-get install --no-install-recommends -y apt-transport-https \
+RUN apt-get update && \
+    apt-get install -y gcc-multilib && \
```
Comment (csujedihy) [inline .docker/ubuntu-22.04/Dockerfile:8]: why do we have a separate line for installing `gcc-multilib`?

File: .docker/ubuntu-22.04/Dockerfile (hunk 1 above)
Comment (ami-GS) [reply .docker/ubuntu-22.04/Dockerfile:67]: yes, for building ebpf program

**Comment by ami-GS** [comment]
What! the behavior differs with my local Ubuntu VM.
Ubuntu VM can not find gcc-multilib, Ubuntu container cannot find ethertool

File: .docker/ubuntu-22.04/Dockerfile (hunk 2 above)
Comment (ami-GS) [reply .docker/ubuntu-22.04/Dockerfile]: My VM could install gcc-multilib before installing rest of packages for some reason...
changing experiment environment to container

**Comment by ami-GS** [comment]
The Build Nuget Package job is gotten affected by different PR?
https://github.com/microsoft/msquic/pull/3819

File: .docker/ubuntu-22.04/Dockerfile [hunk 3]
```diff
@@ -16,11 +16,8 @@ RUN apt-get update && apt-get install --no-install-recommends -y apt-transport-h
     apt-add-repository ppa:lttng/stable-2.13 && \
     apt-get update && apt-get install -y tzdata && apt-get install -y \
     build-essential \
-    gcc-arm-linux-gnueabihf \
```
Comment (csujedihy) [inline .docker/ubuntu-22.04/Dockerfile:19]: why removing these? they are needed for cross compile

Comment (ami-GS) [reply .docker/ubuntu-22.04/Dockerfile:19]: gcc-multilib automatically removes these libraries, but leave these there makes installation failing

Comment (csujedihy) [reply .docker/ubuntu-22.04/Dockerfile:19]: then you can't have gcc-multilib in this docker file or you will need to figure out a way to make it coexist with these libs.

Comment (ami-GS) [reply .docker/ubuntu-22.04/Dockerfile]: found the way to cover in CMakeLists.txt without installing gcc-multilib

**Review by csujedihy — APPROVED** [review]

**Pull request merged**